code/system/application/perception/include/detectlane/detectlane.hpp

```diff
@@ -56,34 +58,50 @@ class DetectLane
  private:
   void setUp();
   void tearDown();
-  odcore::data::dmcp::ModuleExitCodeMessage::ModuleExitCode body();
+  // odcore::data::dmcp::ModuleExitCodeMessage::ModuleExitCode body();
 
   bool m_initialized;
-  cv::Mat m_image;
+  cv::Mat m_currentImg;
+  uint16_t m_blurKernelSize; 
+  cv::Mat m_cannyImg;
+  cv::Mat m_adapThreshImg;
   std::deque<std::pair<odcore::data::TimeStamp, cv::Mat>> m_visualMemory;
-  uint16_t m_intensityThreshold;
+  uint8_t m_adapThreshKernelSize;
+  uint8_t m_adapThreshConst;
   uint16_t m_cannyThreshold;
   uint16_t m_houghThreshold;
+  std::vector<cv::Vec2f> m_linesRaw;
+  std::vector<cv::Vec2f> m_linesProcessed;
+  std::vector<uint16_t> m_laneLineIds;
+  std::vector<uint16_t> m_currentLaneLineIds;
+  std::vector<cv::Vec2f> m_xScreenP;
+  std::vector<cv::Vec2f> m_yScreenP;
+  std::vector<cv::Vec2f> m_xWorldP;
+  std::vector<cv::Vec2f> m_yWorldP;
+  float m_lineDiff;
+  float m_OneLineDiff;
+  float m_HorisontalLimit;
   double m_memThreshold;
   double m_upperLaneLimit;
   double m_lowerLaneLimit;
   int16_t m_screenSize[2];
   int16_t m_roi[4];
   odcore::base::Mutex m_mtx;
   bool m_debug;
+  std::string m_cameraName; 
   Eigen::Matrix3d m_transformationMatrix;
 
-  void GetGrouping(std::vector<cv::Vec2f> &, std::vector<cv::Vec2f> &, double);
-  void GetParametricRepresentation(std::vector<cv::Vec2f> &,std::vector<cv::Vec2f> &,std::vector<cv::Vec2f> &);
-  void GetPointsOnLine(std::vector<cv::Vec2f> &a_xPoints
-    , std::vector<cv::Vec2f> &a_yPoints
-    , std::vector<cv::Vec2f> &a_X
-    , std::vector<cv::Vec2f> &a_Y
-    , std::vector<cv::Vec2f> &a_p
-    , std::vector<cv::Vec2f> &a_m);
-  void GetLinePairs(std::vector<cv::Vec2f> &, std::vector<int8_t> &);
-  Eigen::MatrixXd ReadMatrix(std::string fileName, uint8_t nRows, uint8_t nCols);
-  void TransformPointToGlobalFrame(Eigen::Vector3d &point);
+  bool ExtractSharedImage(odcore::data::image::SharedImage *);
+  void UpdateVisualMemory();
+  void UpdateVisualLines();
+  void GetGrouping(std::vector<cv::Vec2f> &, std::vector<cv::Vec2f>, double);
+  std::vector<std::pair<cv::Vec2f, cv::Vec2f>> GetParametricRepresentation(std::vector<cv::Vec2f>);
+  void UpdatePointsOnLines(std::vector<std::pair<cv::Vec2f, cv::Vec2f>>);
+  std::vector<uint16_t> GetLanes() const;
+  std::vector<uint16_t> GetCurrentLane() const;
+  Eigen::MatrixXd ReadMatrix(std::string const, uint8_t const, uint8_t const);
+  Eigen::Vector3d TransformPointToGlobalFrame(Eigen::Vector3d) const;
+  void DrawWindows();
 
```

Comment:
Variables should always be put last in any scope (e.g. "public", "private"). Please move the methods before the variables.